src/algebra/order/pointwise.lean

```diff
@@ -1,20 +1,106 @@
 /-
 Copyright (c) 2021 Alex J. Best. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Alex J. Best
+Authors: Alex J. Best, Yaël Dillies
 -/
-import algebra.order.field
-import algebra.algebra.basic
+import algebra.bounds
 
 /-!
 # Pointwise operations on ordered algebraic objects
 
 This file contains lemmas about the effect of pointwise operations on sets with an order structure.
 
+## TODO
+
+`Sup (s • t) = Sup s • Sup t` and `Inf (s • t) = Inf s • Inf t` hold as well but
+`covariant_class` is currently not polymorphic enough to state it.
 -/
 
+open function set
 open_locale pointwise
 
+variables {α : Type*}
+
+section conditionally_complete_lattice
+variables [conditionally_complete_lattice α]
+
+section has_one
+variables [has_one α]
+
+@[simp, to_additive] lemma cSup_one : Sup (1 : set α) = 1 := cSup_singleton _
+@[simp, to_additive] lemma cInf_one : Inf (1 : set α) = 1 := cInf_singleton _
+
+end has_one
+
+section group
+variables [group α] [covariant_class α α (*) (≤)] [covariant_class α α (swap (*)) (≤)] {s t : set α}
+
+@[to_additive] lemma cSup_inv (hs₀ : s.nonempty) (hs₁ : bdd_below s) : Sup s⁻¹ = (Inf s)⁻¹ :=
+by { rw ←image_inv, exact ((order_iso.inv α).map_cInf' hs₀ hs₁).symm }
+
+@[to_additive] lemma cInf_inv (hs₀ : s.nonempty) (hs₁ : bdd_above s) : Inf s⁻¹ = (Sup s)⁻¹ :=
+by { rw ←image_inv, exact ((order_iso.inv α).map_cSup' hs₀ hs₁).symm }
+
+@[to_additive] lemma cSup_mul :
+  s.nonempty → bdd_above s → t.nonempty → bdd_above t → Sup (s * t) = Sup s * Sup t :=
```

Comment:
I realise it looks silly since it instantiates arguments and then immediately applies them but I slightly prefer this for consistency with the other lemmas here.
```suggestion
@[to_additive] lemma cSup_mul (hs₀ : s.nonempty) (hs₁ : bdd_above s) (ht₀ : t.nonempty)
  (ht₁ : bdd_above t) :
  Sup (s * t) = Sup s * Sup t :=
```
and likewise for `cInf_mul`.